include/convex_bodies/hpolytope.h

```diff
@@ -638,6 +639,124 @@ class HPolytope {
     }
 
 
+    //------------------------------oracles for exponential sampling---------------//////
+
+    // compute intersection points of a ray starting from r and pointing to v
+    // with polytope discribed by A and b
+    std::pair<NT, int> quadratic_positive_intersect(Point const& r,
+                                    Point const& v,
+                                    VT& Ac,
+                                    NT const& T,
+                                    VT& Ar,
+                                    VT& Av,
+                                    int& facet_prev) const
+    {
+        NT lamda = 0, lamda2 =0, lamda1 =0, alpha;
+        NT min_plus  = std::numeric_limits<NT>::max();
+        NT max_minus = std::numeric_limits<NT>::lowest();
+        VT sum_nom;
+        bool real;
```

Comment:
see comment below of how to avoid using this variable.

Reply:
It is used by function 'solve_quadratic_polynomial'. The comment below is about function `solve_qudratic_polynomial_stable`.

Reply:
I know, but the suggestion is the same. If `solve_quadratic_polynomial` return the `bool` value then the `real` variable is not needed here.

Reply:
Thanks. Done.